udfs/migration/teradata/right.sql

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+CREATE OR REPLACE FUNCTION td.right(string_expr STRING, length_expr INT64) AS (
+  SUBSTR(string_expr, LENGTH(string_expr) - length_expr + 1, length_expr)
```

Comment:
The value of `LENGTH(string_expr) - length_expr + 1` will become negative if the `length_expr` is  > the length of the `string_expr`. This causes the `SUBSTR` function to begin truncating from the end of the string again which is unexpected behavior. Perhaps the following would be better.

```sql
CASE
  WHEN length_expr < 0 THEN ERROR('The length argument to RIGHT must be a positive integer!')
  ELSE SUBSTR(string_expr, -1 * length_expr, length_expr)
END
```